ospfd/ospf_zebra.c

```diff
@@ -1619,27 +1612,19 @@ void ospf_distribute_list_update(struct ospf *ospf, int type,
 				 unsigned short instance)
 {
 	struct ospf_external *ext;
-	void **args = XCALLOC(MTYPE_OSPF_DIST_ARGS, sizeof(void *) * 2);
-
-	args[0] = ospf;
-	args[1] = (void *)((ptrdiff_t)type);
 
 	/* External info does not exist. */
 	ext = ospf_external_lookup(ospf, type, instance);
-	if (!ext || !EXTERNAL_INFO(ext)) {
-		XFREE(MTYPE_OSPF_DIST_ARGS, args);
+	if (!ext || !EXTERNAL_INFO(ext))
 		return;
-	}
 
 	/* If exists previously invoked thread, then let it continue. */
-	if (ospf->t_distribute_update) {
-		XFREE(MTYPE_OSPF_DIST_ARGS, args);
+	if (ospf->t_distribute_update)
```

Comment:
this test is redundant right?  thread_add_timer_msec eventually calls into code that checks that the thread isn't alreadyscheduled.

Reply:
Removed.